rules-reviewed/azure/eap/eap-to-azure-appservice-environment-variables.windup.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0"?>
+<ruleset id="eap-to-azure-appservice-environment-variables"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Validate that all usages of System.getenv and System.getProperty are captured
+            as App Settings when migrating to Azure App Service.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[7,8)" />
+        <targetTechnology id="azure-appservice"/>
+        <tag>JBoss EAP</tag>
+        <tag>Azure</tag>
+    </metadata>
+    <rules>
+        <rule id="eap-to-azure-appservice-environment-variables-001">
+            <when>
+                <or>
+                    <!-- as compiled code -->
+                    <javaclass references="java.lang.System.getenv({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <javaclass references="java.lang.System.getProperty({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <!-- as source code -->
+                    <filecontent filename="{*}.java" pattern="System.getenv"/>
+                    <filecontent filename="{*}.java" pattern="System.getProperty"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="App Service - The application reads environment variables" category-id="optional" effort="1">
+                    <message>
+                        <![CDATA[
+                        App Service - The application reads environment variables.
+
+                        Any environment variables used in the code will need to be defined as App Settings, which are exposed
+                        as environment variables in App Service.
+    
+                        Any System properties that the code depends on will need to be provided either in a custom Startup Script
+                        or by defining an App Setting with the name JAVA_OPTS and include any system properties in it.
+                        ]]>
+                    </message>
+
+                    <link title="Customization and tuning" href="https://aka.ms/webapps-java-perf"/>
+                    <link title="Environment variables and app settings in Azure App Service" href="https://aka.ms/webapps-env-vars"/>
+                </hint>
+                <technology-tag level="INFORMATIONAL">Azure</technology-tag>

```

Comment:
```suggestion
```
I think this can be removed as there's already the `<tag>Azure</tag>` above and the `technology-tag` is meant to identify technologies found in the analyzed application while, in this case, `Azure` is the target technology so not already adopted in the application.